src/main/java/org/lisoft/lsml/view_fx/controls/EquippedItemCell.java

```diff
@@ -318,22 +331,60 @@ private boolean onRemoveItem() {
         return safeCommand(this, stack, new CmdRemoveItem(msgd, loadout, component, getItem()), msgd);
     }
 
-    private void setupEngineRatingDropDown(boolean aPgiMode) {
+    private <T> boolean addIfEngineExists(final ObservableList<T> aList, final T aValue, final Integer aRating, final EngineType aType, final Faction aFaction) {
```

Comment:
Return value is never used, you can remove it. The arguments: `aValue` and `aRating` are the same, drop one and take the argument as `int` which avoids the null check. Faction can never be null as this dialog is never reached without a loadout and makes no sense without a loadout, so you can drop that null check as well.

Reply:
Yeah, this was vestigial code from when I made the type dropdown regenerate as well, but then I realized that making it so that XL doesn't show up in the dropdown when you have STD 60 selected was dumb.

What I really wanted this to look like in the first place was something along the lines of the scala `(engineMin to engineMax by 5).filter(rating => engineExists(type, rating))`, but I gave up when I saw that IntStream doesn't let you create a finite stream that steps by a value that isn't 1 without doing something horrible like `IntStream.range(min/step,max/step).map(i -> i * step)`.

Renamed to addRatingIfEngineExists, and replaced the null checks with Objects.requiresNonNull  to be a bit more explicit (let me know if you want me to delete these as well).